runtime/src/main/java/com/flipkart/flux/resource/ClientElbResource.java

```diff
@@ -0,0 +1,175 @@
+package com.flipkart.flux.resource;
+
+import com.codahale.metrics.annotation.Timed;
+import com.flipkart.flux.api.ClientElbDefinition;
+import com.flipkart.flux.domain.ClientElb;
+import com.flipkart.flux.representation.ClientElbPersistenceService;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Named;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.UUID;
+
+/**
+ * API for CRUD operations on Client Cluster ELB details
+ * @author akif.khan
+ */
+@Singleton
+@Path("/api/clientelb")
+@Named
+public class ClientElbResource {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClientElbResource.class);
+
+    private ClientElbPersistenceService clientElbPersistenceService;
+
+    @Inject
+    public ClientElbResource(ClientElbPersistenceService clientElbPersistenceService
+    ) {
+        this.clientElbPersistenceService = clientElbPersistenceService;
+
+    }
+
+    /**
+     * Create and persist new client's ELB details viz. ElbURL and ClientId
+     *
+     * @param clientId Unique Id specific to a Client Cluster
+     * @param clientElbUrl Client ELB URL Address
+     */
+    @POST
+    @Path("/create")
+    @Timed
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createClientElb(@QueryParam("clientId") String clientId,
+                                @QueryParam("clientElbUrl") String clientElbUrl) {
+        if(clientElbUrl == null)
+            return Response.status(Response.Status.BAD_REQUEST.getStatusCode()).entity(
+                    "ClientElbUrl cannot be null").build();
+        else {
+            try {
+                URL verifyingURL = new URL(clientElbUrl);
+                verifyingURL.toURI();
+                if(clientId == null)
+                    clientId = UUID.randomUUID().toString();
```

Comment:
why is this random? client-id should be mandatory param for the API.

Reply:
Made client-id field mandatory, no more randomly generated clientIds' .